### ISSUE TITLE
Reducing streaming costs over extended periods of time

### ISSUE THREAD

**Issue by sholzmayer** [chengsokdara/use-whisper#20]
Currently, the streaming feature works perfectly fine, but it sends the entire audio stream from the beginning based on the timeSlice seconds. This results in exponential costs with longer periods. For instance, recording around 15 minutes can cost up to $10 with a timeSlice of 1 second.

To avoid such high costs, I suggest implementing a new feature that would resend only the last n-seconds of the audio stream. This would provide some context while reducing the amount of seconds being sent and thus, lowering the costs.

I believe that this improvement would not only make the streaming feature more cost-effective but also enhance its overall performance. 

In the attached screenshot you can see the API usage from a 15 minutes streaming transcription:

<img width="434" alt="Screenshot 2023-03-26 at 19 49 47" src="https://user-images.githubusercontent.com/20296626/227794430-328c953c-0151-4186-8696-682dc6524ce5.png">

**Comment by MatthewMariner** [comment]
I also have the same issue - basically can't use this library (even though I love it!) until this is fixed as the rendering costs make it unaffordable.

**Comment by grtzsohalf** [comment]
I will appreciate a lot if this issue is solved!

**Comment by Aslanf8** [comment]
I agree! This would be a fantastic addition. The cost exponentially grows as time continues.

**Comment by 0arm** [comment]
How about this. Use the browser speech transcription API's to provide immediate feedback for transcription. But then after the the recording is done. Send the final audio file to openai whisper api. And once you have the better transcript, transition from the browser ASR result to the final result in the UI

**Comment by shawhu** [comment]
i was developing a very similar feature and i hit a wall. i was trying to send delta data (each chunk) to the whisper api and it didn't work. and i tried to play it locally, also didn't work. that's how i found this repo.
basically it seems that when you do this

    voiceRecorder.ondataavailable = ({ data }) => {
       //trying to use this data
    }

only the first blob can be played. all the rest of them are missing some metadata or something. the only way to make those playable is to add them together. like if you have 3 of them, you will have to add data0+data1+data2 together to create a new blob and create a file from that blob, just like the author did in the code.

**Comment by taha-yva** [comment]
Anyone who figured out solution of this problem.

**Comment by m-fraczek** [comment]
@shawhu I'm stuck with the same exact issue, have you found a way to send the blobs separately?
